python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -9070,6 +9070,149 @@ def test_dataframe_add_suffix():
     assert_eq(got, expected)
 
 
+@pytest.mark.parametrize(
+    "data, gkey",
+    [
+        (
+            {
+                "id": ["a", "a", "a", "b", "b", "b", "c", "c", "c"],
+                "val1": [5, 4, 6, 4, 8, 7, 4, 5, 2],
+                "val2": [4, 5, 6, 1, 2, 9, 8, 5, 1],
+                "val3": [4, 5, 6, 1, 2, 9, 8, 5, 1],
+            },
+            ["id", "val1", "val2"],
+        ),
+        (
+            {
+                "id": [0] * 4 + [1] * 3,
+                "a": [10, 3, 4, 2, -3, 9, 10],
+                "b": [10, 23, -4, 2, -3, 9, 19],
+            },
+            ["id", "a"],
+        ),
+        (
+            {
+                "id": ["a", "a", "b", "b", "c", "c"],
+                "val": cudf.Series(
+                    [None, None, None, None, None, None], dtype="float64"
+                ),
```

Comment:
I think this is equivalent in behavior and will avoid the need to construct data on the GPU during test collection (takes considerable time). `@pytest.mark.parametrize` is called during collection, so providing a `cudf.Series` or `cudf.DataFrame` as a parameter makes the startup time for pytest longer even if we don't run the test.
```suggestion
                "val": [np.nan, np.nan, np.nan, np.nan, np.nan, np.nan],
```

Reply:
With your suggestion ,the test fails : `TypeError: Covariance accepts only numerical column-pairs`, which is why Michael suggested using cudf Series to cast the Null columns to a numerical dtype (float). 
Any other workaround ideas?

Reply:
This is because by default you have `nan_as_null` flag on. Something we are not very sure atm... But you should achieve similar result by setting that flag to `False`.